### PR TITLE
Adding netcore sample of showcase

### PR DESCRIPTION
**Pull request by WojcikMike**
This PR requires 2 more things to be covered:
 - Managing version of signalr.js (at the moment it's copied into the lib with the latest version pulled from npm
 - It has commented out debug flags as the state in signalr core was removed, So I need to change it so that the value is passed as a parameter.

### CODE REVIEW

File: samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "iisSettings": {
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Properties/launchSettings.json]: This entire file isn't needed right?

File: samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Startup.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+            endpoint = Endpoint.Start(endpointConfiguration).GetAwaiter().GetResult();
+
+            return new AutofacServiceProvider(container);
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Startup.cs]: Add shutdown similar to https://github.com/Particular/docs.particular.net/pull/3797

File: samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Store.ECommerce.Core.csproj [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  <ItemGroup>
+    <None Include="wwwroot\product.js" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Alexinea.Autofac.Extensions.DependencyInjection" Version="4.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.1.0-rc1-final" />
+	<PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
+    <PackageReference Include="NServiceBus" Version="7.*" />
+    <PackageReference Include="NServiceBus.Autofac" Version="7.0.0-rc0001" />
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Store.ECommerce.Core.csproj]: This should be the final version right?

**Comment by WojcikMike** [comment]
@andreasohlund I addressed your comments

**Comment by andreasohlund** [comment]
LGTM, ping when you want a final review and merge

File: samples/showcase/on-premises/Core_7/Store.Sales/Store.Sales.Core.csproj [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+	<LangVersion>latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>    
+    <PackageReference Include="Newtonsoft.Json" Version="10.*" />
+    <PackageReference Include="NServiceBus" Version="7.*" />
+    <PackageReference Include="NServiceBus.Encryption.MessageProperty" Version="2.*" />
+	<ProjectReference Include="..\Store.Messages\Store.Messages.Core.csproj" />
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.Sales/Store.Sales.Core.csproj]: Nitpick: spacing issue

**Comment by WojcikMike** [comment]
So i fixed the debug flag. In ASPNETCore it is a parameter in signalr calls, as the hub context was removed.

When it goes with signalr.js file, I will keep this as is (js file in repo rather than some packaging solution). In that case sample will be easier to run. What is more for Signalr to work the versions need to match between server and the client, and apparently the ones that work together now are: 
2.1-rc1-final on the server side and 1.0.0-rc1-update1 for js signalr.

**Comment by WojcikMike** [comment]
This PR is ready to me reviewed

**Comment by WojcikMike** [comment]
@Particular/deployment-maintainers @bording to make this PR build we need .net core sdk 2.1.300-rc1 installed on the agents. could you please do it

**Comment by bording** [comment]
@WojcikMike The official RTM bits should be available today or tomorrow, so let's just wait for those instead of installing the RC.

File: samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Properties/launchSettings.json (hunk 1 above)
Comment (andreasohlund) [reply samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Properties/launchSettings.json]: Ping, is this still needed?

File: samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Store.ECommerce.Core.csproj [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="wwwroot\product.js" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Alexinea.Autofac.Extensions.DependencyInjection" Version="4.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.1.0-rc1-final" />
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.ECommerce.Core/Store.ECommerce.Core.csproj]: With 2.1 out does this need to change?

File: samples/showcase/on-premises/Core_7/Store.Sales/Store.Sales.Core.csproj [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+	<LangVersion>latest</LangVersion>
+  </PropertyGroup>
+  <ItemGroup>    
+    <PackageReference Include="Newtonsoft.Json" Version="10.*" />
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.Sales/Store.Sales.Core.csproj]: Should be 11.*

File: samples/showcase/on-premises/Core_7/Store.Shared/Store.Shared.Core.csproj [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Library</OutputType>
+    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <ApplicationIcon />
+    <StartupObject />
+  </PropertyGroup>
+  <ItemGroup>    
+    <PackageReference Include="Newtonsoft.Json" Version="10.*" />
```
Comment (andreasohlund) [inline samples/showcase/on-premises/Core_7/Store.Shared/Store.Shared.Core.csproj]: Should be 11.*

**Comment by WojcikMike** [comment]
@andreasohlund I addressed your comments

File: samples/showcase/on-premises/Core_7/Store.ContentManagement/Store.ContentManagement.Core.csproj [hunk 8]
```diff
@@ -5,8 +5,7 @@
     <TargetFramework>netcoreapp2.0</TargetFramework>
 	<LangVersion>latest</LangVersion>
   </PropertyGroup>
-  <ItemGroup>
```
[the remark this answers is not in the document]
Comment (WojcikMike) [reply samples/showcase/on-premises/Core_7/Store.ContentManagement/Store.ContentManagement.Core.csproj]: It is. I was inspired by: azure variation of this sample: https://github.com/Particular/docs.particular.net/blob/master/samples/showcase/cloud-azure/Core_7/Store.CustomerRelations/Store.CustomerRelations.csproj 
Which is obviously violating this rule. Fixing it right now

**Pull request merged**